### PR TITLE
Validating transaction signature for send transaction methods

### PR DESCRIPTION
**Pull request by rmoreliovlabs**
## Description
Add more security to our send transaction methods

## Motivation and Context
As of now, we validate tx signature only in the [eth_sendTransaction](https://github.com/rsksmart/rskj/blob/cea83a74da1caf5761c806eaad740fd19b6c7b90/rskj-core/src/main/java/co/rsk/rpc/modules/eth/EthModuleTransactionBase.java#L79-L81). We should also do this in the personal_sendTransaction and eth_sendRawTransaction methods.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist:

- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Requires Activation Code (Hard Fork)

* **Other information**:

### CODE REVIEW

File: rskj-core/src/main/java/co/rsk/rpc/modules/eth/EthModuleTransactionBase.java [hunk 1]
```diff
@@ -106,6 +106,16 @@ public String sendRawTransaction(String rawData) {
 				throw invalidParamError("Missing parameter, gasPrice, gas or value");
 			}
 
+			Account senderAccount = this.wallet.getAccount(tx.getSender());
+
+			if (senderAccount == null) {
+				throw RskJsonRpcRequestException.invalidParamError(TransactionArgumentsUtil.ERR_COULD_NOT_FIND_ACCOUNT + tx.getSender().toHexString());
+			}
```
Comment (Vovchyk) [inline rskj-core/src/main/java/co/rsk/rpc/modules/eth/EthModuleTransactionBase.java:113]: we don't need this particular sender account check in the `sendRawTransaction()` method, as the sender account doesn't really need to be part of the `wallet` in this context.

File: rskj-core/src/main/java/co/rsk/rpc/modules/personal/PersonalModuleWalletEnabled.java [hunk 2]
```diff
@@ -199,6 +203,10 @@ private String sendTransaction(CallArguments args, Account senderAccount) throws
 
 		tx.sign(senderAccount.getEcKey().getPrivKeyBytes());
 
+        if (!tx.acceptTransactionSignature(constants.getChainId())) {
+            throw RskJsonRpcRequestException.invalidParamError(TransactionArgumentsUtil.ERR_INVALID_CHAIN_ID + tx.getChainId());
+        }
```
Comment (Vovchyk) [inline rskj-core/src/main/java/co/rsk/rpc/modules/personal/PersonalModuleWalletEnabled.java:208]: please fix the indentation here

**Review by Vovchyk — CHANGES_REQUESTED** [review]
- could you cover the new logic with unit tests?
- please fix the LGTM alert

File: rskj-core/src/main/java/co/rsk/rpc/modules/eth/EthModuleTransactionBase.java (hunk 1 above)
Comment (rmoreliovlabs) [reply rskj-core/src/main/java/co/rsk/rpc/modules/eth/EthModuleTransactionBase.java]: Done

File: rskj-core/src/main/java/co/rsk/rpc/modules/personal/PersonalModuleWalletEnabled.java (hunk 2 above)
Comment (rmoreliovlabs) [reply rskj-core/src/main/java/co/rsk/rpc/modules/personal/PersonalModuleWalletEnabled.java:208]: Ran intellij formatter for both files

**Comment by rmoreliovlabs** [comment]
pipeline:run

File: rskj-core/src/main/java/co/rsk/rpc/modules/eth/EthModuleTransactionBase.java [hunk 3]
```diff
@@ -41,85 +41,89 @@
...
+            if (null == tx.getGasLimit() || null == tx.getGasPrice() || null == tx.getValue()) {
+                throw invalidParamError("Missing parameter, gasPrice, gas or value");
+            }
 
-			TransactionPoolAddResult result = transactionGateway.receiveTransaction(tx);
-			if (!result.transactionsWereAdded()) {
-				throw RskJsonRpcRequestException.transactionError(result.getErrorMessage());
-			}
+            if (!tx.acceptTransactionSignature(constants.getChainId())) {
+                throw RskJsonRpcRequestException.invalidParamError(TransactionArgumentsUtil.ERR_INVALID_CHAIN_ID + tx.getChainId());
```
Comment (Vovchyk) [inline rskj-core/src/main/java/co/rsk/rpc/modules/eth/EthModuleTransactionBase.java:110]: please cover this thrown exception in unit tests

File: rskj-core/src/main/java/co/rsk/rpc/modules/personal/PersonalModuleWalletEnabled.java [hunk 4]
```diff
@@ -187,22 +191,26 @@ private Account getAccount(String from, String passphrase) {
...
 
-		TransactionArguments txArgs = TransactionArgumentsUtil.processArguments(args, transactionPool, senderAccount, config.getNetworkConstants().getChainId());
+        Transaction tx = Transaction.builder().withTransactionArguments(txArgs).build();
 
-		Transaction tx = Transaction.builder().withTransactionArguments(txArgs).build();
+        tx.sign(senderAccount.getEcKey().getPrivKeyBytes());
 
-		tx.sign(senderAccount.getEcKey().getPrivKeyBytes());
+        if (!tx.acceptTransactionSignature(constants.getChainId())) {
+            throw RskJsonRpcRequestException.invalidParamError(TransactionArgumentsUtil.ERR_INVALID_CHAIN_ID + tx.getChainId());
```
Comment (Vovchyk) [inline rskj-core/src/main/java/co/rsk/rpc/modules/personal/PersonalModuleWalletEnabled.java:207]: please cover this thrown exception in unit tests

**Review by Vovchyk — CHANGES_REQUESTED** [review]

File: rskj-core/src/main/java/co/rsk/rpc/modules/personal/PersonalModuleWalletEnabled.java (hunk 4 above)
Comment (rmoreliovlabs) [reply rskj-core/src/main/java/co/rsk/rpc/modules/personal/PersonalModuleWalletEnabled.java:207]: Done

File: rskj-core/src/main/java/co/rsk/rpc/modules/eth/EthModuleTransactionBase.java (hunk 3 above)
Comment (rmoreliovlabs) [reply rskj-core/src/main/java/co/rsk/rpc/modules/eth/EthModuleTransactionBase.java:110]: done

**Review by Vovchyk — APPROVED** [review]

**Comment by rmoreliovlabs** [comment]
pipeline:run

**Comment by rmoreliovlabs** [comment]
pipeline:run

**Comment by rmoreliovlabs** [comment]
pipeline:run

**Comment by rmoreliovlabs** [comment]
pipeline:run

**Review by illuque-iov — APPROVED** [review]

File: rskj-core/src/test/java/co/rsk/test/builders/TransactionBuilder.java [hunk 5]
```diff
@@ -127,10 +139,12 @@
 
     /**
      * Generates a random transaction
-     * */
+     */
     public Transaction buildRandomTransaction() {
-        int i = new Random().nextInt();
-        BigInteger randomPositiveVal = i > 0 ?  BigInteger.valueOf(i) : BigInteger.valueOf(i * -1);
+        long i = new Random().nextInt();
```
[the remark this answers is not in the document]
Comment (rmoreliovlabs) [reply rskj-core/src/test/java/co/rsk/test/builders/TransactionBuilder.java:144]: Solved

**Comment by Vovchyk** [comment]
pipeline:run

**Review by Vovchyk — APPROVED** [review]

**Pull request merged**